nautobot/docs/development/apps/api/platform-features/prepopulating-data.md

```diff
@@ -0,0 +1,125 @@
+
+## Advanced
+
+While the signal handler approach works for most use cases, it does have its downsides. Other options for the population of base state are available, each with their own trade-offs. There is no single best option, you have to choose the one has the best properties for your specific need.
+
+- Signal Handlers on `nautobot_database_ready` (see above)
+- [Django Onboard Tooling](https://docs.djangoproject.com/en/4.2/howto/initial-data/)
+    - [Django Data Migrations](https://docs.djangoproject.com/en/4.2/topics/migrations/#data-migrations)
+    - [Django Fixtures](https://docs.djangoproject.com/en/4.2/topics/db/fixtures/#fixtures)
+- [Design Builder Jobs](https://docs.nautobot.com/projects/design-builder/en/latest/)
+- Creating data in-place where needed
+
+The following table details points that are somewhat easily comparable between the different options.
+
+| Feature           | Signal Handlers on `nautobot_database_ready`                                         | Django Data Migrations               | Django Fixtures                                                                      | Design Builder Jobs                                                                  | Creating Data In-place Where Needed                                                  |
+|-------------------|--------------------------------------------------------------------------------------|--------------------------------------|--------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------|
+| Startup Impact    | Exceptions prevent container startup                                                 | Exceptions prevent container startup | N/A                                                                                  | N/A                                                                                  | N/A                                                                                  |
+| Execution         | Automatic                                                                            | Automatic                            | Manual                                                                               | Manual                                                                               | Automatic                                                                            |
+| Performance       | Average                                                                              | Average                              | Good                                                                                 | Average                                                                              | Average                                                                              |
```

Comment:
Four "Average" and one "Good" doesn't provide a lot of differentiation. :) Perhaps:

```suggestion
| Performance Impact | Each time migrations are rerun | Once when migrations are first run | Once when fixture is loaded | Each time design job is run | Each time the data is needed |
```